### PR TITLE
UIOR-864 Check for potential duplicate order lines

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UIOR-864

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/LayerCollection/LayerPOLine.js [hunk 1]
```diff
@@ -184,13 +190,61 @@ function LayerPOLine({
     }
   };
 
+  const validateDuplicateLines = useCallback(
```
Comment (NikitaSedyx) [inline src/components/LayerCollection/LayerPOLine.js:193]: can it be moved from component?

**Review by usavkov-epam — APPROVED** [review]

File: src/components/LayerCollection/LayerPOLine.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/components/LayerCollection/LayerPOLine.js]: it should + we have a ticket to use this one in open order functionality. Thanx

File: src/common/utils/validateDuplicateLines.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { batchFetch } from '@folio/stripes-acq-components';
+import { VALIDATION_ERRORS } from '../constants';
+
+export const validateDuplicateLines = (line, mutator) => {
```
Comment (NikitaSedyx) [inline src/common/utils/validateDuplicateLines.js:4]: not sure it's common util

File: src/common/utils/validateDuplicateLines.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  return batchFetch(mutator.poLines, productIds, (itemsChunk) => {
+    const query = itemsChunk
+      .map(productId => `details.productIds=="*${productId}*"`)
+      .join(' and ');
+
+    return query
+      ? `${baseQuery} and (${query})`
+      : baseQuery;
+  }).then((existingLines) => {
```
Comment (NikitaSedyx) [inline src/common/utils/validateDuplicateLines.js:16]: from what I see we follow style or async/await

```
promise
  .then
```

File: src/common/utils/validateDuplicateLines.js (hunk 2 above)
Comment (NikitaSedyx) [reply src/common/utils/validateDuplicateLines.js:4]: ignore, we have ticket for open as well

File: src/common/utils/validateDuplicateLines.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/common/utils/validateDuplicateLines.js:16]: We have different styles even in one function) Should I rewrite this function with async/await?

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**